client/src/controllers/ZoneController.test.js

```diff
@@ -0,0 +1,137 @@
+import { Application } from '@hotwired/stimulus';
+import { ZoneController } from './ZoneController';
+
+describe('ZoneController', () => {
+  let application;
+
+  const setup = async (html) => {
+    document.body.innerHTML = `<main>${html}</main>`;
+
+    application = Application.start();
+    application.register('w-zone', ZoneController);
+
+    await Promise.resolve();
+  };
+
+  afterEach(() => {
+    application?.stop();
+    jest.clearAllMocks();
+  });
+
+  describe('connect method', () => {
+    it('should dispatch a "ready" event when connected', async () => {
+      const readyHandler = jest.fn();
+      document.addEventListener('w-zone:ready', readyHandler);
+
+      await setup(`
+        <div
+          data-controller="w-zone"
+        ></div>
+      `);
```

Comment:
```suggestion
      await setup(`
        <div data-controller="w-zone"></div>
      `);
```

Small nitpick, I know some auto-formatters do this but for this small HTML snippet, let's just keep it on one line.